masonry/src/text/render_text.rs

```diff
@@ -60,6 +61,63 @@ pub fn render_text(
                         }
                     }),
                 );
+            // We want to draw the underline/strikethrough on top of the glyphs, so we draw it later
+            if let Some(underline) = &style.underline {
+                let underline_brush = &brushes[underline.brush.0];
+                let run_metrics = glyph_run.run().metrics();
+                let offset = match underline.offset {
+                    Some(offset) => offset,
+                    None => run_metrics.underline_offset,
+                };
+                let width = match underline.size {
+                    Some(size) => size,
+                    None => run_metrics.underline_size,
+                };
+                // The `offset` is the distance from the baseline to the *top* of the underline
+                // so we move the line down by half the width
+                // Remember that we are using a y-down coordinate system
+                let y = glyph_run.baseline() - offset + width / 2.;
```

Comment:
```suggestion
                let y = glyph_run.baseline() - offset.;
```

I don't think we should ever render underlines above the defined offset... they should always extend downwards.

Reply:
Firefox also only extends downwards.

Reply:
Hmm, I'm struggling to understand this suggestion. The only way I can make the two halves of this comment agree is if we are:
1) actually in a y-up coordinate space; OR
2) that width is negative

Both of which I believe to be false.
Do you agree that adding a positive value in a y-down space leads to a movement downwards?

Reply:
Yes. Your code (for the underline) is correct, I think? The stroke expands both up and down by width, so you have to offset it down (positive value) by half the width to keep its top at the same place.

Reply:
Agreed, I've restored it